zipkin-autoconfigure/storage-elasticsearch-http/src/main/java/zipkin/autoconfigure/storage/elasticsearch/http/ZipkinElasticsearchHttpStorageAutoConfiguration.java

```diff
@@ -59,4 +65,17 @@ StorageComponent storage(ElasticsearchHttpStorage.Builder esHttpBuilder) {
       return hosts != null && (hosts.contains("http://") || hosts.contains("https://"));
     }
   }
-}
+
+  static final class isBasicAuthRequired implements Condition {
+    @Override
+    public boolean matches(ConditionContext condition, AnnotatedTypeMetadata annotatedTypeMetadata) {
+      String userName = condition.getEnvironment().getProperty("zipkin.storage.elasticsearch.basic-auth-user-name");
```

Comment:
please simplify to "basic-auth-user-name" to just "username" and same for password (the property name is overly specific as we won't have any other approach)